src/main/java/uk/gov/hmcts/reform/sscs/servicebus/TopicPublisher.java

```diff
@@ -36,21 +39,29 @@ public TopicPublisher(JmsTemplate jmsTemplate,
         maxAttempts = 5,
         backoff = @Backoff(delay = 2000, multiplier = 3)
     )
-    public void sendMessage(final String message) {
-        log.info("Sending message.");
+    public void sendMessage(final String message, String caseId, final AtomicReference<Message> msg) {
+        log.info("Sending message for caseId {}", caseId);
+
         try {
-            jmsTemplate.send(destination, (Session session) -> session.createTextMessage(message));
-            log.info("Message sent.");
+            jmsTemplate.convertAndSend(destination, message, m -> {
```

Comment:
Why it has been change from existing session.createTextMessage(message)

Reply:
The messageId gets set by the message broker and not by the publisher. It's not known before. The `convertAndSend` method allows a postProcessor argument which is where we are able to pass the updated message into the `msg` atomic reference on line 48/49. We can't get access to the messageId via createTextMessage method

Reply:
Will it create any issue to publish and read message by other services?